pkg/build/sbom.go

```diff
@@ -0,0 +1,68 @@
+// Copyright Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"istio.io/pkg/log"
+	"istio.io/release-builder/pkg/model"
+	"istio.io/release-builder/pkg/util"
+)
+
+// Sbom generates Software Bill Of Materials for istio repo in an SPDX readable format.
+func GenerateBillOfMaterials(manifest model.Manifest) error {
+	// Retrieve istio repository path to run the sbom generator
+	istioRepoDir := manifest.RepoDir("istio")
+	sourceSbomFile := path.Join(manifest.OutDir(), "istio-source.spdx")
+	sourceSbomNamespace := fmt.Sprintf("https://storage.googleapis.com/istio-release/releases/%s/istio-source.spdx",
```

Comment:
Is this the correct URL? It does make sense to have the published URL, but I see that as something like `https://gcsweb.istio.io/gcs/istio-release/releases/1.12.2/`, but that doesn't seem to map to the storage.googleapis.com.

Reply:
Given URL works for manifest : https://storage.googleapis.com/istio-release/releases/1.12.2/manifest.yaml and since spdx files are published at the same location as manifest.yaml, I am assuming this would be the right URL once spdx files are published.